internal/versions/versions.go

```diff
@@ -175,42 +175,103 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
 
 // Returns the OsImage entity
 func (h *handler) GetOsImage(openshiftVersion, cpuArchitecture string) (*models.OsImage, error) {
+	if cpuArchitecture == "" {
+		// Empty implies default CPU architecture
+		cpuArchitecture = common.DefaultCPUArchitecture
+	}
+	// Filter OS images by specified CPU architecture
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
+		if swag.StringValue(osImage.CPUArchitecture) == "" {
+			return cpuArchitecture == common.DefaultCPUArchitecture
+		}
+		return swag.StringValue(osImage.CPUArchitecture) == cpuArchitecture
+	})
+	if funk.IsEmpty(osImages) {
+		return nil, errors.Errorf("The requested CPU architecture (%s) isn't specified in OS images list", cpuArchitecture)
+	}
+
+	// Search for specified x.y.z openshift version
+	osImage := funk.Find(osImages, func(osImage *models.OsImage) bool {
+		return swag.StringValue(osImage.OpenshiftVersion) == openshiftVersion
+	})
+
 	versionKey, err := h.getKey(openshiftVersion)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, osImage := range h.osImages {
-		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
-			cpuArchitecture = common.DefaultCPUArchitecture
-		}
-		if *osImage.OpenshiftVersion == versionKey && *osImage.CPUArchitecture == cpuArchitecture {
-			return osImage, nil
+	if osImage == nil {
+		// Fallback to x.y version
+		osImage = funk.Find(osImages, func(osImage *models.OsImage) bool {
+			return *osImage.OpenshiftVersion == versionKey
+		})
+	}
+
+	if osImage == nil {
+		// Find latest available patch version by x.y version
+		osImages := funk.Filter(osImages, func(osImage *models.OsImage) bool {
+			imageVersionKey, err := h.getKey(*osImage.OpenshiftVersion)
+			if err != nil {
+				return false
+			}
+			return imageVersionKey == versionKey
+		}).([]*models.OsImage)
+		sort.Slice(osImages, func(i, j int) bool {
+			return *osImages[i].OpenshiftVersion > *osImages[j].OpenshiftVersion
```

Comment:
Does this direct comparison work?
Wouldn't `4.10` end up sorting the opposite of expected compared to `4.9` when using straight string or numerical sort?

Reply:
Can you make sure we have a test for this with `4.9.1`, `4.9.9`, and `4.9.10` versions?